R/app_alphaDiv.R

```diff
@@ -0,0 +1,96 @@
+#' Alpha diversity
+#'
+#' This function returns alpha diversity values for each sample.
+#' 
+#' @param otu data.frame with samples as rows, taxa as columns
+#' @param method string defining the the alpha diversity method. Accepted values are 'shannon','simpson', and 'evenness'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return something that's useful. TBD
+#' @importFrom vegan diversity
+#' @importFrom stringi stri_trans_totitle
+#' @import veupathUtils
+#' @import data.table
+#' @export
+alphaDiv <- function(otu, method = c('shannon','simpson','evenness'), verbose = c(TRUE, FALSE)) {
+
+    # Initialize and check inputs
+    method <- veupathUtils::matchArg(method)
+    verbose <- veupathUtils::matchArg(verbose)
+
+    computeMessage <- ''
+    veupathUtils::logWithTime(paste("Received OTU table with", NROW(otu), "samples and", (NCOL(otu)-1), "taxa."), verbose)
+    
+
+    # Compute alpha diversity
+    if (identical(method, 'shannon') | identical(method, 'simpson')){
+
+      alphaDivDT <- try(vegan::diversity(otu[, -c('SampleID')], method))
```

Comment:
dont think we can assume a column name here. youll have to be told what it is somehow, and will take the form of 'entity.var' like all others. except the var will be 'blah_id' or something